pirania/files/usr/bin/voucher

```diff
@@ -104,7 +118,44 @@ captive_portal.status = function(context)
     print ( logic.status(db, mac) )
 end
 
--- List all
+-- Show numbers of active vouchers
+captive_portal.show_active_vouchers = function()
+    local result = {}
+    local vName = 1
+    local vExpire = 3
+    local usedMacs = 7
+    members = 0
+    visitors = 0
+    local db = dba.load(config.db)
+    for _, v in pairs(db.data) do
+        local expireDate = tonumber(v[vExpire]) or 0
+        local active = function ()
+            if (#v[usedMacs] > 0) then
+                return true
+            end
+            return false
+        end
+        if (expireDate > dateNow() and active() == true) then
+            local nameInfo = {}
+            for word in v[vName]:gmatch("[^-]+") do
+                table.insert(nameInfo, word)
+            end
+            if (nameInfo[1] == 'm') then
+                local nextN = members + 1
+                members = nextN
+            else
+                local nextN = visitors + 1
+                visitors = nextN
+            end
+        end
+        result.members = members
+        result.visitors = visitors
+    end
+    printJson(result)
+end
+
+
+-- List all vouchers
```

Comment:
I would try to move most of the code of this file into a library , like (` pirania/files/usr/lib/lua/...`). This way we would be able to test and share code and is easier to maintain than having code in the "binary folders".